OpenSim/Examples/CustomActuatorExample/toyLeg_example.cpp

```diff
@@ -65,33 +65,35 @@ int main()
         Inertia linkageInertia = Inertia::cylinderAlongY(linkageDiameter/2.0, linkageLength/2.0);
 
         OpenSim::Body* linkage1 = new OpenSim::Body("linkage1", linkageMass, linkageMassCenter, linkageMass*linkageInertia);
-
-        // Graphical representation
-        linkage1->addDisplayGeometry("cylinder.vtp");
-        //This cylinder.vtp geometry is 1 meter tall, 1 meter diameter.  Scale and shift it to look pretty
-        GeometrySet& geometry = linkage1->updDisplayer()->updGeometrySet();
-        DisplayGeometry& thinCylinder = geometry[0];
-        thinCylinder.setScaleFactors(linkageDimensions);
-        thinCylinder.setTransform(Transform(Vec3(0.0,linkageLength/2.0,0.0)));
-        linkage1->addDisplayGeometry("sphere.vtp");
-        //This sphere.vtp is 1 meter in diameter.  Scale it.
-        geometry[1].setScaleFactors(Vec3(0.1));
         
+        // Graphical representation
+        Cylinder cyl;
+        cyl.set_scale_factors(linkageDimensions);
+        Frame* cyl1Frame = new PhysicalOffsetFrame(*linkage1, Transform(Vec3(0.0, linkageLength / 2.0, 0.0)));
+        cyl1Frame->setName("Cyl1_frame");
+        osimModel.addFrame(cyl1Frame);
```

Comment:
While I agree that `adopt` is more accurate and I am happy that our containers have the adopt and append semantics, when it comes to building a model programmatically, I think it is less intuitive to user developers and scripting users than `addWhatever()`. I think you should file an issue so we can discuss it. 

If we had a generic `Component::add<ComponentType>(ComponentType& comp)` it would eliminate the need for individual add methods.  We could eliminate Sets by instead enabling Component to build/curate its own internal lists based on whatever type is being added. It would still be fast to find and access specific components via a `get/upd<ComponentType>(name)` capability. This seems both more compact and scalable than having `addBody`, `addJoint`, `addConstraint`, `addForce`, `addProbe`, `addFrame`, `addGeometry` which is unsustainable!

I am also not a fan of the current `addWhatever(Whatever* thingy)` form which takes a pointer to newed Whatever. We could reference count (perhaps via a ComponentHandle) so that Components being added can always be allocated on the stack (like with MobilizedBody) by the user.